### PR TITLE
vendor: bump pebble to 560af6e1

### PR DESCRIPTION
**Pull request by sumeerbhola**
560af6e1 db: fix bug in seek noop optimization and add more testing

Release note: None

Release justification: A small bug fix to a Pebble change that
was reverted earlier this day. The Pebble change significantly
improves performance with separated intents as indicated in
the original Pebble PR
https://github.com/cockroachdb/pebble/pull/1067 and is a
prerequisite for enabling separated intents (Category 2 or 4)

### CODE REVIEW

**Review by sumeerbhola — COMMENTED** [review]
I am currently running `roachtest run "acceptance/bank/cluster-recovery" --count 100 --cloud gce`. I was not able to reproduce https://github.com/cockroachdb/cockroach/issues/60828, so it is remotely possible that there is a different bug. The change to `iterator.go` was small, so I think the risk of a different bug is low.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/60956#-:-MUAp2Z8DehNWM90qiP9:b-rfog7s)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @itsbilal, @jbowens, and @nvanbenschoten)

**Review by itsbilal — APPROVED** [review]
:lgtm:

> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/60956#-:-MUAuqLV8VUCuo1WFvBv:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @jbowens and @nvanbenschoten)

**Comment by sumeerbhola** [comment]
acceptance/bank/cluster-recovery failed again.
I am so far unable to reproduce using Pebble tests -- the Pebble metamorphic test had run for > 4h before merging the Pebble change. I will make changes to that test to see if I can trigger a failure.

**Comment by sumeerbhola** [comment]
> acceptance/bank/cluster-recovery failed again.
I am so far unable to reproduce using Pebble tests -- the Pebble metamorphic test had run for > 4h before merging the Pebble change. I will make changes to that test to see if I can trigger a failure.

update:
- hacked up the Pebble metamorphic test to add SingleDeletes and to use MVCCKeys, and unable to reproduce after many hours of runs.
- ran master with acceptance/bank/cluster-recovery and extra invariant checking in intentInterleavingIter and the key mangling unsafeMVCCIterator, in case the bug is not in Pebble but something to do with the integration in CockroachDB. 100 runs with no failures.

**Comment by sumeerbhola** [comment]
https://github.com/cockroachdb/cockroach/pull/61194 and https://github.com/cockroachdb/pebble/pull/1073 will unblock bumping Pebble. I have run acceptance/bank/cluster-recovery 500 times with no failure.

**Pull request closed without merging**
